Source/Core/Core/State.cpp

```diff
@@ -485,8 +513,23 @@ u64 GetUnixTimeOfSlot(int slot)
 
 static void LoadFileStateData(const std::string& filename, std::vector<u8>& ret_data)
 {
-  Flush();
-  File::IOFile f(filename, "rb");
+  File::IOFile f;
+
+  {
+    // If a state is currently saving, wait for that to end or time out.
+    std::unique_lock lk(s_state_writes_in_queue_mutex);
+    if (s_state_writes_in_queue != 0)
+    {
+      if (!s_state_write_queue_is_empty.wait_for(lk, std::chrono::milliseconds(100),
+                                                 []() { return s_state_writes_in_queue == 0; }))
+      {
+        Core::DisplayMessage(
```

Comment:
Not sure how likely this issue will occur and maybe this is more work than it is worth but I wonder if a "Waiting for in progress saving operation to finish..." with a "cancel" button would be more user friendly?